File: typescript-client-sdk/src/sdk/models/operations/getallcrmnotes.ts
import { SpeakeasyMetadata, SpeakeasyBase } from "../../../internal/utils";
import * as shared from "../shared";



export class GetAllCrmNotesQueryParams extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "queryParam, style=form;explode=true;name=accessToken" })
  accessToken: string;

  @SpeakeasyMetadata({ data: "queryParam, style=form;explode=true;name=allFields" })
  allFields?: boolean;

  @SpeakeasyMetadata({ data: "queryParam, style=form;explode=true;name=cursor" })
  cursor?: string;

  @SpeakeasyMetadata({ data: "queryParam, style=form;explode=true;name=limit" })
  limit?: number;
}


export class GetAllCrmNotesResponseBody extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "json, name=nextPageCursor" })
  nextPageCursor?: string;

  @SpeakeasyMetadata({ data: "json, name=notes", elemType: shared.Note })
  notes?: shared.Note[];
}


export class GetAllCrmNotesRequest extends SpeakeasyBase {
  @SpeakeasyMetadata()
  queryParams: GetAllCrmNotesQueryParams;
}


export class GetAllCrmNotesResponse extends SpeakeasyBase {
  @SpeakeasyMetadata()
  contentType: string;

  @SpeakeasyMetadata()
  statusCode: number;

  @SpeakeasyMetadata()
  responseBody?: GetAllCrmNotesResponseBody;
}

File: go-client-sdk/pkg/models/operations/getallcrmnotes.go
package operations

import (
	"github.com/speakeasy-sdks/orb-sdks/go-client-sdk/pkg/models/shared"
)

type GetAllCrmNotesQueryParams struct {
	AccessToken string   `queryParam:"style=form,explode=true,name=accessToken"`
	AllFields   *bool    `queryParam:"style=form,explode=true,name=allFields"`
	Cursor      *string  `queryParam:"style=form,explode=true,name=cursor"`
	Limit       *float64 `queryParam:"style=form,explode=true,name=limit"`
}

type GetAllCrmNotesResponseBody struct {
	NextPageCursor *string       `json:"nextPageCursor,omitempty"`
	Notes          []shared.Note `json:"notes,omitempty"`
}

type GetAllCrmNotesRequest struct {
	QueryParams GetAllCrmNotesQueryParams
}

type GetAllCrmNotesResponse struct {
	ContentType  string
	StatusCode   int64
	ResponseBody *GetAllCrmNotesResponseBody
}
